[CAP- 90-91] Campus Toggle Button

damle22 opened the pull request:
This pull request introduces a toggle button to switch between the SGW and LOY campuses on the map. The button updates the displayed location dynamically by restarting the MapsActivity with the new campus coordinates.

**Key Features:**
✅ Toggle between SGW and LOY campuses
✅ Updates map location upon switch
✅ Preserves toggle state on activity reload
✅ Uses Intent to pass new coordinates and campus name

**Changes Implemented:**

1. Added a button (campusSwitchBtn) to switch between campuses.

2. Implemented toggleCampus() method to handle campus switching logic.

3. Used Intent extras to reload MapsActivity with updated coordinates.

4. Ensured campus name is displayed dynamically (campusTextView).

**How It Works:**

1. Clicking the button switches between SGW and LOY campuses.

2. The app reloads MapsActivity with the new location.

3. The correct campus name and coordinates are retained

damle22 committed:
implemented the logic for toggling between campuses

zachariyaJaveri commented on app/src/main/java/minicap/concordia/campusnav/screens/MapsActivity.java:
```diff
@@ -40,11 +59,19 @@ protected void onCreate(Bundle savedInstanceState) {
+        campusTextView = findViewById(R.id.ToCampus);
+        campusTextView.setText(campusNotSelected);
+
+        campusSwitchBtn = findViewById(R.id.campusSwitch);
```
Can you move this to the subscribeButtons function that we have? Just to ensure that all our subscription code is in one place

damle22 replied:
the subscribeButtons function is located in MainActivity.java -> the above piece of code is located in MapsActivity.java and needs to be called inside this screen

zachariyaJaveri commented on app/src/main/java/minicap/concordia/campusnav/screens/MapsActivity.java:
```diff
@@ -58,6 +85,26 @@ protected void onCreate(Bundle savedInstanceState) {
+        // Flipping the building state
+        showSGW = !showSGW;
+        // Reloading MapsActivity with the new building coordinates
+        Intent i = new Intent(MapsActivity.this, MapsActivity.class);
```
I don't think it's a good idea to start a new activity every time we click the toggle button, this will consume more ram and is not necessary, just change the coordinates that are centered on the map

damle22 replied:
Yeah you are right on this. I will fix it by rather changing coordinates on map and center the camera on it. Working on it.

zachariyaJaveri commented on app/src/main/java/minicap/concordia/campusnav/screens/MapsActivity.java:
```diff
@@ -24,14 +28,29 @@ public class MapsActivity extends FragmentActivity implements OnMapReadyCallback
+    public static final String KEY_CAMPUS_NOT_SELECTED = "campus_not_selected";
     private static final int LOCATION_PERMISSION_REQUEST_CODE = 1;
 
+    private final LatLng SGW_LOCATION = new LatLng(45.49701, -73.57877);
```
I don't think this is necessary, you can use the CoordResHelper to get this information without having to store it a second time, same for below

zachariyaJaveri commented on build.gradle.kts:
```diff
@@ -1,6 +1,6 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application") version "8.8.0" apply false
+    id("com.android.application") version "8.2.2" apply false
```
Was this changed by gradle? If so, please revert

zachariyaJaveri reviewed (commented):
Needs refactoring and changes (see comments)

damle22 committed:
fixed pull request issues

Made sure to not call the activity but rather update the camera & marker position.

Kayram2710 reviewed (approved):
Good functionality, good job. WIll resolve merge conflicts.

Kayram2710 committed:
Merge into main
